compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -203,6 +203,14 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
             }
             sym::unlikely => self
                 .call_intrinsic("llvm.expect.i1", &[args[0].immediate(), self.const_bool(false)]),
+            sym::select_unpredictable => {
+                let cond = args[0].immediate();
+                let true_val = args[1].immediate();
+                let false_val = args[2].immediate();
+                let result = self.select(cond, true_val, false_val);
```

Comment:
Note that this currently only handles immediates, which is sufficient for binary search.